tests/suites/host_test.function

```diff
@@ -682,8 +683,8 @@ int execute_tests( int argc , const char ** argv )
             // If there are no unmet dependencies execute the test
             if( unmet_dep_count == 0 )
             {
-                test_info.result = TEST_RESULT_SUCCESS;
-                test_info.step = (unsigned long)( -1 );
+                mbedtls_test_info.result = MBEDTLS_TEST_RESULT_SUCCESS;
```

Comment:
What about to add an mbedtls_test_info_reset() to do this and reset to zero the other fields ?
With this the `memset` @604 can probably be removed.

Reply:
Yes I personally I like the idea of adding an API to get/set `mbedtls_test_info` so that `mbedtls_test_info` could be kept local and private to `helpers.c`. The reason that I didn't do so was because there are lots of places (such as L#90-95) where each field needs to be individually set and I didn't much like the idea of having 5 different getters and 5 different setters to access each field as needed.

Ideally I would have used a [variadic template](https://en.wikipedia.org/wiki/Variadic_template) (C++) but as that is not possible in C I am unsure what the best course of action is. Do you think it is worth just having a "reset" and "increment" for now and possibly expanding that in future or should this all be left to another PR/Issue to handle?

Reply:
The improvement now would be just to NOT set fields of mbedtls_test_info directly in target_test.function and host_test.function. It is ok to set the fields directly in helpers.c and to read then directly in helpers.c and *.function files.

Reply:
That change sounds sensible to me. I will make the change soon.

Reply:
Done.